src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.cs

```diff
@@ -30,154 +21,14 @@ protected AbstractEditorInlineRenameService(IEnumerable<IRefactorNotifyService>
             _globalOptions = globalOptions;
         }
 
-        protected abstract bool CheckLanguageSpecificIssues(
-            SemanticModel semantic, ISymbol symbol, SyntaxToken triggerToken, [NotNullWhen(true)] out string? langError);
-
         public async Task<IInlineRenameInfo> GetRenameInfoAsync(Document document, int position, CancellationToken cancellationToken)
         {
-            var triggerToken = await GetTriggerTokenAsync(document, position, cancellationToken).ConfigureAwait(false);
-
-            if (triggerToken == default)
-            {
-                return new FailureInlineRenameInfo(EditorFeaturesResources.You_must_rename_an_identifier);
-            }
-
-            return await GetRenameInfoAsync(_refactorNotifyServices, document, triggerToken, cancellationToken).ConfigureAwait(false);
-        }
-
-        private async Task<IInlineRenameInfo> GetRenameInfoAsync(
-            IEnumerable<IRefactorNotifyService> refactorNotifyServices,
-            Document document, SyntaxToken triggerToken,
-            CancellationToken cancellationToken)
-        {
-            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
-            if (syntaxFacts.IsReservedOrContextualKeyword(triggerToken))
-                return new FailureInlineRenameInfo(EditorFeaturesResources.You_must_rename_an_identifier);
```

Comment:
The new code will reference hte strings in the Features layer.  And it will return just an instance of SymbolicRenameInfo, not a subclasss.